cmd/containerd/command/main.go

```diff
@@ -281,6 +281,18 @@ can be used and modified as necessary as a custom configuration.`
 		if err != nil {
 			return fmt.Errorf("failed to get listener for main endpoint: %w", err)
 		}
+		if config.GRPC.AddressHardLink != "" {
+			if _, err := os.Stat(config.GRPC.AddressHardLink); err == nil {
+				os.Remove(config.GRPC.AddressHardLink)
+			}
+			// Ensure parent directory is created
+			if err = os.MkdirAll(filepath.Dir(config.GRPC.AddressHardLink), 0770); err != nil {
+				return err
+			}
+			if err = os.Link(config.GRPC.Address, config.GRPC.AddressHardLink); err != nil {
```

Comment:
Hmm. you can link by yourself instead of adding code like this.
If you're using systemd to manage containerd life-cycle, you can use `ExecStartPost= ` in systemd file (with Type=notify).

REF: https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#ExecStartPre=

Reply:
thanks  @fuweid